### PR TITLE
Update consumer logic with timeOut

### PR DESCRIPTION
**Pull request by ketamethane**
1. Update DLQ logic
2. Current Timeout function works without use of TTL and requires the use of promises for error handling. Upon rejecting a message, message is moved to DLQ where the user will then be notified about rejection
3. `matchService.js` service is currently not in use and has been removed as the logic has all moved to the `subscriber.js` file
4. Remove assertExchange() in publisher logic

### CODE REVIEW

File: Backend/MatchingService/rabbitmq/subscriber.js [hunk 1]
```diff
@@ -1,41 +1,207 @@
...
 
 // To remember what goes in a subscriber use some Acronym
 // Connect, Assert, Process, E - for Acknowledge
 
+const dead_letter_queue_name = "dead_letter_queue";
+const timeoutMap = {};
+// Local dictionary to store waiting users
+const waitingUsers = {};
+
+// manual rejection of message that is constantly being consumed. Not using TTL feature.
```
Comment (RingoftheKing) [inline Backend/MatchingService/rabbitmq/subscriber.js]: Is this segment considered dead code? If so we might want to remove it now.

**Review by RingoftheKing — COMMENTED** [review]
.

File: Backend/MatchingService/rabbitmq/setup.js [hunk 2]
```diff
@@ -10,71 +13,50 @@ const queueNames = [
...
 
         if (!connection) {
             return;
         }
 
         const channel = await connection.createChannel();
 
-        // Declare matching exchange to be bind to queues 
-
+        // Declare the matching exchange (topic)
```
Comment (ketamethane) [inline Backend/MatchingService/rabbitmq/setup.js:28]: Note to self: remove line 12 in publisher (assertExchange)

File: Backend/MatchingService/rabbitmq/subscriber.js (hunk 1 above)
Comment (ketamethane) [reply Backend/MatchingService/rabbitmq/subscriber.js]: Removed

**Comment by yuxuanleong** [comment]
Matching and time out works fine

**Pull request merged**